### PR TITLE
[Intl] Blacklist Eurozone and United Nations in Region Data Generator

### PR DESCRIPTION
**Pull request by gregurco**
| Q             | A
| ------------- | ---
| Branch?       | 2.8
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #28260 
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Review by nicolas-grekas — APPROVED** [review]

File: src/Symfony/Component/Intl/Data/Generator/RegionDataGenerator.php [hunk 1]
```diff
@@ -34,6 +34,8 @@ class RegionDataGenerator extends AbstractDataGenerator
     const BOUVET_ISLAND_ID = 'BV';
     const HEARD_MCDONALD_ISLANDS_ID = 'HM';
     const CLIPPERTON_ISLAND_ID = 'CP';
+    const EUROZONE_UNION_ID = 'EZ';
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Intl/Data/Generator/RegionDataGenerator.php]: actually, this should be named EUROZONE_ID I suppose

Comment (gregurco) [reply src/Symfony/Component/Intl/Data/Generator/RegionDataGenerator.php]: fixed. I found in wiki that EZ is a **monetary** union, that's why I added union in constant name :slightly_smiling_face:  but yep, finally it's not a usual "union".

**Comment by nicolas-grekas** [comment]
Actually, the data also had to be updated. I just did it by pushing to your fork.

**Review by javiereguiluz — APPROVED** [review]

**Review by stof — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @gregurco.

**Pull request merged**